src/catalogue/infrastructure/adapter/DatabaseProduitAdapter.ts

```diff
@@ -8,24 +8,32 @@ import {
 } from "../../usecases/recupererLesProduits/filtreProduit";
 import {AucunProduitTrouve} from "../../domain/errors/AucunProduitTrouve";
 import http2 from "http2";
+import {Connection} from "typeorm";
+import TypeORMClient from "../../../configuration/database/TypeORMClient";
+import ProduitTypeORMEntity from "../../configuration/db/type-orm-entity/produitTypeORMEntity";
 
-export default class ProduitAdapter implements ProduitPort {
+export default class DatabaseProduitAdapter implements ProduitPort {
   listeProduit: Produit[]
 
-  constructor() {
+  constructor(private typeORMClient: TypeORMClient) {
     this.listeProduit = []
   }
 
-  sauvegarderProduit(produit: Produit): Produit {
-    this.listeProduit.push(produit)
-    return produit
+  async sauvegarderProduit(produit: Produit): Promise<void> {
+    await this.typeORMClient.executeQuery(db => db.getRepository(ProduitTypeORMEntity).save(ProduitTypeORMEntity.fromProduit(produit)))
   }
 
-  récupérerLesProduits(filtre?: FiltreProduit): Array<Produit> {
+  async récupérerLesProduits(filtre?: FiltreProduit): Promise<Produit[]> {
```

Comment:
mettre to do tri
virer le get pour avoir une vraie methode
commande adapter : renommer mapper
ajouter les retours